### DISCUSSION TITLE
Remix: submission fails only in Chrome

### DISCUSSION THREAD

**Discussion by dvnrsn** [edmundhung/conform#534] [General]
This is super random and I am not sure what to make of it or where to put it

```js
    const submission = parseWithZod(formData, {schema: formSchema});

    if (submission.status !== 'success') {
      return submission.reply();
    }
```

I see that we fail `submission.status !== success` and return submission.reply() but only in Chrome (we've tested incognito and vice versa)

The value of submission.reply()

```js
{
  status: undefined,
  intent: { type: 'validate', payload: { name: 'email' } },
  initialValue: {
    email: 'gibberish@gibberish.com',
    form_type: 'inline-form',
    full_name: 'Gibberish Gibero',
    list_id: 'Ifj2'
  },
  error: undefined,
  state: { validated: { full_name: true, email: true } }
}
```

The result is that the browser redirects to the action route itself resulting in the display of JSON.

When I console submission itself I see

```js
{
  status: undefined,
  payload: {
    list_id: 'XzfCUy',
    form_type: 'inline-form',
    full_name: 'Giberish',
    email: 'gibberish@gibberish.com'
  },
  error: {},
  reply: [Function: reply]
}

**Comment by dvnrsn** [comment]
I believe this is a double submit as I see two POSTs in the dev tools but only in Chrome.

Note that if I remove `{...getFormProps(form)}` the submission works as expected

**Reply by edmundhung** [reply]
Any chance you are making a form submit yourself on the `onSubmit` handler? You will need to call `event.preventDefault()` if you wanna cancel the default form submission.

**Reply by dvnrsn** [reply]
Hmm, no we're using fetcher.Form

```js
<fetcher.Form
  method="post"
  action="/our/path"
  {...getFormProps(form)}
>
```
It's so bizarre to me how this is a Chrome specific issue on this page (we're able to see other forms using conform working on other pages). Fortunately we can get away without the browser validation for this form.

**Reply by edmundhung** [reply]
Just notice you mentioned this above
> The result is that the browser redirects to the action route itself resulting in the display of JSON.

If this happens, it's very likely there is an error thrown during onSubmit and so Remix fallback to native form submission with the document reloaded (that's why it is showing the action data as JSON). There should be errors logged on console when this happened. Or you can try catching the error manually like this:

```tsx
<fetcher.Form
  method="post"
  action="/our/path"
  {...getFormProps(form)}
  onSubmit={event => {
    try {
      form.onSubmit(event);
    } catch (e) {
      console.log(e);
    }
  }}
>
```

**Reply by dvnrsn** [reply]
Good thought, thanks. Sadly this is all we get (notice the double POST), even with the console in the onSubmit callback. 

<img width="531" alt="image" src="https://github.com/edmundhung/conform/assets/9112811/65c0d698-24b0-4a40-9280-b6624ed13811">

I'll have to try moving to another route or something if we come back to this.

**Reply by edmundhung** [reply]
Can you show me the payload of the two POST requests? I am wondering if this is related to #524 which seems to happens on chromium based browser only

**Reply by dvnrsn** [reply]
Certainly. Here is the first form data payload

```yml
full_name: devin
email: devin@devin.com
__state__: {"validated":{"full_name":true}}
```

And second:

```yml
full_name: devin
email: devin@devin.com
__state__: {"validated":{"full_name":true}}
__intent__: {"type":"validate","payload":{"name":"email"}}
```


edit: 
What I would expect is one of the following.

```js
{"validated":{"full_name":true,"email":true}}
```

```yaml
__state__: {"validated":{}}
```

This is what we see in the cases that it works.

FWIW I don't think we're in a Dialog and removing `shouldValidate: 'onBlur'` (i.e., going to onSubmit) does seem to help

**Reply by edmundhung** [reply]
I think the issue you had is likely fixed on [v1.0.6](https://github.com/edmundhung/conform/releases/tag/v1.0.6).

**Reply by dvnrsn** [reply]
Sadly it doesn't seem like this helps

```js
    "@conform-to/react": "^1.0.6",
    "@conform-to/zod": "^1.0.6",
```

I did notice that once it works it works until I hard refresh and use new email. This immediately triggers the undesired redirect (still only in Chrome)

**Comment by deadcoder0904** [comment]
@dvnrsn do you think [my issue & yours is related](https://github.com/edmundhung/conform/discussions/527)? i can't get 1st failure error in ui even if its shown in console.

**Reply by dvnrsn** [reply]
There are two things I'm observing that seem very similar.

1. submitting the form via return key correctly triggers the UI change in Firefox but _not_ in Chrome (unless incognito strangely)
2. commenting out `shouldValidate: 'onBlur'` seems to help

Can you confirm either of these?
